.circleci/config.yml

```diff
@@ -50,15 +65,33 @@ jobs:
           command: docker-compose run base sbt "project webknossosDatastore" compile stage
 
       - save_cache:
-          key: cache-{{ checksum ".circleci/cache_version" }}-{{ .Branch }}-{{ .Revision }}
+          key: target-cache-{{ checksum ".circleci/cache_version" }}-{{ .Branch }}-{{ .Revision }}
           paths:
             - "project/target"
             - "target"
+
+      - save_cache:
+          key: sbt-cache-{{ checksum ".circleci/cache_version" }}-{{ checksum "project/Dependencies.scala" }}
+          paths:
             - "~/.m2"
             - "~/.ivy2"
             - "~/.sbt"
+
+      - save_cache:
+          key: yarn-cache-{{ checksum ".circleci/cache_version" }}-{{ checksum "yarn.lock" }}
+          paths:
             - "~/.yarn-cache"
 
+      - save_cache:
+          key: webpack-uglifyjs-cache-{{ checksum ".circleci/cache_version" }}-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - "node_modules/.cache/uglifyjs-webpack-plugin"
```

Comment:
Yeah, I think I would have done that if there were more "coherently connected", but in addition to the fact that the webpack cache has to be restored **after** yarn install, I would leave it as isolated as it is now. I added name attributes to the caching steps, though, so that the summary is more readable.